### ISSUE TITLE
Middle deletes corrupt the tree.

### ISSUE THREAD

**Issue by SergeyFromHell** [qwertie/btree-typescript#2]
Seems that Btree become corrupted (forgots some keys) after some operations.
Here is an example:

```
const BTree = require('sorted-btree').default;

const tree = new BTree();
const keys = new Set();

// tree and keys should always contain same keys
for (let i=0; ; i++) {
    // add to both
    tree.set(i, i);
    keys.add(i);
    if (tree.size > 25000 && i % 2 == 0) {
        const key = Math.floor(i / 2);
        // delete from both
        tree.delete(key);
        keys.delete(key);
        for (let key of keys) {
            if (!tree.has(key)) {
                console.error('No key ' + key + ' found in tree at cycle', i);
                process.exit(1);
            }
        }
    }
}
```

Use version 1.2.1 of sorted-btree and Node 13.5.0.

**Comment by qwertie** [comment]
I've published a fix as _sorted-btree@1.2.2_ with a unit test. If the `BTree` gives you any trouble in the future, please call `btree.checkValid()` after any suspect operations. This function scans the tree for invalid state and throws an `Error` if anything is wrong.

**Comment by qwertie** [comment]
OMG, this was submitted on Jan 14? I'm so sorry, somehow I overlooked this for the longest time until your extra messages came in. Mind you, when I finally did run the code, no error occurred, because `for (let key of keys)` doesn't work as you mentioned.
